decorum/decorum.py

```diff
@@ -26,40 +29,40 @@ def __init__(self, *args, **kwargs):
         #: directly to the matching attributes on the decorator.
         self.assigned = functools.WRAPPER_ASSIGNMENTS
         if 'assigned' in kwargs:
-            self.assigned = kwargs['assigned']
+            self.assigned = kwargs.pop('assigned')
 
         if args and callable(args[0]):
             # used as decorator without being called
             self.init()
-            self._wrapped = self.__call__(args[0])
+            self.wraps(args[0])
         else:
             # used as decorator and called
             self.init(*args, **kwargs)
 
-    def __call__(self, f=None, *args, **kwargs):
+    def __call__(self, *args, **kwargs):
         """Actually run the decorated function.
 
         Uses :meth:`wrap` to handle decoration. Restores :attr:`__doc__` and
         :attr:`__name__`.
 
         """
-        if not callable(f):
-            if f:
-                return self._wrapped(f, *args, **kwargs)
-            else:
-                return self._wrapped(*args, **kwargs)
+        if self._wrapped:
+            return self.call(*args, **kwargs)
         else:
-            wrapped = self.wraps(f)
-            return wrapped
+            f = args[0]
+            return self.wraps(f)
 
     def wraps(self, f):
         """Wraps the function and returns it"""
+        self._wrapped = f
```

Comment:
I wonder about moving this line in both `__init__` and `__call__`, so that, if users forget to call `super(subclass, self).wraps(function)`, they lose the update_wrapper feature, but still have the `self._wrapped` attribute.